cupy/sorting/search.py

```diff
@@ -181,7 +182,132 @@ def where(condition, x=None, y=None):
     return _where_ufunc(condition.astype('?'), x, y)
 
 
-# TODO(okuta): Implement searchsorted
+# This is to allow using the same kernels for all dtypes, ints & floats
+# as nan is a special case
+_preamble = '''
+template<typename T>
+__device__ bool _isnan(T val) {
+    return false;
+}
+template<>
+__device__ bool _isnan<float16>(float16 val) {
```

Comment:
I think the template notation is not necessary? Ditto for other specializations.

```suggestion
__device__ bool _isnan(float16 val) {
```

Reply:
Let me try!

Reply:
If we don't use templates, do we have to manually overload the function for all the integer types?
If so, probably is better to just add isnan() for integers in numpy.

Reply:
You can keep the generic template version.